src/esp32cam.h

```diff
@@ -29,33 +42,194 @@ class CameraClass
   bool
   changeResolution(const Resolution& resolution, int sleepFor = 500);
 
+  /** \brief Change camera resolution.
+   *  \pre Change Camera Frequence Clock
+   *  \param iMhz new Frequence, must be in MHZ
+   */
+  bool
+  changeFrequence(int iMhz);
+
+  /** \brief Change Sensor Contrast level
+  * must be -2 to +2, default 0
+  */
+  bool
+  changeContrast(int ilevel);
+
+  /** \brief Change Brightness level
+   *  \param ilevel must be -2 to +2, default 0
+   */
+  bool
+  changeBrightness(int ilevel);
+
+  /** \brief Change Saturation level
+   *  \param ilevel must be -2 to +2, default 0
+   */
+  bool 
+  changeSaturation(int ilevel);
+
+  /** \brief Change SpecialEffect
+   *  \param ilevel must be > 0 && <= NUM_SPECIAL_EFFECTS, default 0
+   *  \	1 = no effect
+   *  \ 2 = negative
+   *  \ 3 = black and white
+   *  \ 4 = reddish
+   *  \ 5 = greenish
+   *  \ 6 = blue
+   *  \ 7 = retro
+   */
+  bool
+  changeSpecialEffect(int ilevel);
+
+  /** \brief Change WbMode
+   *  \param ilevel must be > 0 && <= NUM_WB_MODES, default 0
+   *  \ 0 = default
+   *  \ 1 = sunny
+   *  \ 2 = cloudy
+   *  \ 3 = office
+   *  \ 4 = home
+   */
+  bool
+  changeWbMode(int ilevel);
+
+  /** \brief Change AE Levels
+   *  \param ilevel must be -2 to +2, default 0
+   */
+  bool
+  changeAeLevels(int ilevel);
+  
+  /** \brief Enable/Disable  AEC (Auto Exposure Control)
+   *  \param iEnable must be 0(disable) or 1 (enable)
+   */
+  bool
+  changAutoExposurecontrol(int iEnable);
+
+  /** \brief Change AGC Gain
+   *  \param ilevel must be >= 0 and <= 30, default 30
+   */
+  bool
+  changAgcGain(int ilevel);
+
+  /** \brief Change Gainceiling
+   *  \param iGainCeiling must be between GAINCEILING_2X and GAINCEILING_128X
+   */
+  bool
+  changGainceilingSensor(int iGainCeiling);
```

Comment:
Wouldn't that be `changeGainCeilingSensor` to match the `iGainCeiling` parameter?

Can `GAINCEILING_nX` be any integer between 2 and 128, in which case shouldn't it just be an integer? If it's an enum, change the parameter description to reference the enum?

And maybe drop `Sensor` entirely? Having sensor in the name is either redundant because it's implicit that you're manipulating some of the sensor settings, or it's wrong because it suggests that there is some kind of a sensor that measures something about the gain, like there might be temperature sensor.

Reply:
Possible that some settings did not work at all.
I wanted to have as many sensor options available for testing as possible,